### PR TITLE
Reverts @ConditionalOnBean in SQL autoconfig

### PR DESCRIPTION
**Pull request by joaoandremartins**
Even though tests are passing, my test app is erroring out at runtime.

### CODE REVIEW

**Comment by joaoandremartins** [comment]
We might want to file a new issue after this PR is merged so we get to fix the issue with the lack of a `CloudSqlJdbcInfoProvider` bean.

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java [hunk 1]
```diff
@@ -56,7 +55,6 @@
 @Import({AppEngineJdbcInfoProviderAutoConfiguration.class,
 		MySqlJdbcInfoProviderAutoConfiguration.class,
 		PostgreSqlJdbcInfoProviderAutoConfiguration.class})
-@ConditionalOnBean(CloudSqlJdbcInfoProvider.class)
```
Comment (artembilan) [inline spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java]: How about if move this to the `cloudSqlDataSourceProperties` `@Bean` level?

Comment (joaoandremartins) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java]: That works! I should've thought of that...

**Comment by joaoandremartins** [comment]
This is still not good enough, as now `GcpCloudSqlAutoConfiguration` is always kicking in and enabling `DataSourceProperties`...

If nothing else, we can always control this with a flag, but I wish there was a better way...

**Comment by artembilan** [comment]
I'm not sure in your concern.

The `cloudSqlDataSourceProperties` is marked now with the `@ConditionalOnBean(CloudSqlJdbcInfoProvider.class)` which can be there only in these case:

- the `AppEngineJdbcInfoProviderAutoConfiguration`, but only if we meet a couple conditions on that class;
- the `MySqlJdbcInfoProviderAutoConfiguration` if meet its conditions;
- the `PostgreSqlJdbcInfoProviderAutoConfiguration` if meet its conditions again;
- and finally if end-user provides his/her own `CloudSqlJdbcInfoProvider` bean.

What else are we missing?

**Comment by joaoandremartins** [comment]
The issue is that if I don't have `spring.cloud.gcp.sql.enabled=false` in `application.properties`, `GcpCloudSqlAutoConfiguration` is enabled, enables `DataSourceProperties` and then

`Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.autoconfigure.jdbc.DataSourceProperties]: Constructor threw exception; nested exception is java.lang.NoClassDefFoundError: org/springframework/jdbc/datasource/embedded/EmbeddedDatabaseType`

This happens for a project that is intended to use e.g., Config only.

Since the bean conditional is now in the method, this autoconfig is always enabled. Unless `enabled=false` is provided, since if omitted, `matchIfMissing=true` brings it on.

**Comment by artembilan** [comment]
??? 
```
java.lang.NoClassDefFoundError: org/springframework/jdbc/datasource/embedded/EmbeddedDatabaseType
```
How is it possible at all to live without `spring-jdbc` on classpath?

That's why I say: #371

**Comment by artembilan** [comment]
FYI: #377

**Pull request merged**